Exercise 2 Multiplexer/multiplexer-rt/src/com/tridiumuniversity/multiplexer/BMultiplexer.java

```diff
@@ -0,0 +1,241 @@
+
+////////////////////////////////////////////////////////////////
+// Property "in3"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code in3} property.
+   * @see #getIn3
+   * @see #setIn3
+   */
+  public static final Property in3 = newProperty(0, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code in3} property.
+   * @see #in3
+   */
+  public boolean getIn3() { return getBoolean(in3); }
+  
+  /**
+   * Set the {@code in3} property.
+   * @see #in3
+   */
+  public void setIn3(boolean v) { setBoolean(in3, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "s0"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code s0} property.
+   * @see #getS0
+   * @see #setS0
+   */
+  public static final Property s0 = newProperty(Flags.SUMMARY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code s0} property.
+   * @see #s0
+   */
+  public boolean getS0() { return getBoolean(s0); }
+  
+  /**
+   * Set the {@code s0} property.
+   * @see #s0
+   */
+  public void setS0(boolean v) { setBoolean(s0, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "s1"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code s1} property.
+   * @see #getS1
+   * @see #setS1
+   */
+  public static final Property s1 = newProperty(Flags.SUMMARY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code s1} property.
+   * @see #s1
+   */
+  public boolean getS1() { return getBoolean(s1); }
+  
+  /**
+   * Set the {@code s1} property.
+   * @see #s1
+   */
+  public void setS1(boolean v) { setBoolean(s1, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Property "out"
+////////////////////////////////////////////////////////////////
+  
+  /**
+   * Slot for the {@code out} property.
+   * @see #getOut
+   * @see #setOut
+   */
+  public static final Property out = newProperty(Flags.READONLY, ((BBoolean)(BBoolean.DEFAULT)).getBoolean(), null);
+  
+  /**
+   * Get the {@code out} property.
+   * @see #out
+   */
+  public boolean getOut() { return getBoolean(out); }
+  
+  /**
+   * Set the {@code out} property.
+   * @see #out
+   */
+  public void setOut(boolean v) { setBoolean(out, v, null); }
+
+////////////////////////////////////////////////////////////////
+// Type
+////////////////////////////////////////////////////////////////
+  
+  @Override
+  public Type getType() { return TYPE; }
+  public static final Type TYPE = Sys.loadType(BMultiplexer.class);
+
+/*+ ------------ END BAJA AUTO GENERATED CODE -------------- +*/
+
+    @Override
+    public void changed(Property property, Context context) {
+        super.changed(property, context);
+        if(!property.equals(out)){
```

Comment:
It's a bit safer to do `if (!out.equals(property))`. The reason is that if `null` is passed to this method for the `property` argument, calling `property.equals` will produce a null pointer exception. Since we know that `out` is non-null (we defined it!), calling `out.equals` is a bit safer. This scenario is unlikely since the framework is in charge of passing arguments to this method, but just thought I'd point it out.